frontend/pages/hosts/details/DeviceUserPage/DeviceUserPage.tsx

```diff
@@ -0,0 +1,279 @@
+import FleetIcon from "../../../../../assets/images/fleet-avatar-24x24@2x.png";
+
+const baseClass = "device-user";
+
+interface IDeviceUserPageProps {
+  params: Params;
+}
+
+interface IHostResponse {
+  host: IHost;
+  org_logo_url: string;
+}
+
+const DeviceUserPage = ({
+  params: { device_auth_token },
+}: IDeviceUserPageProps): JSX.Element => {
+  const deviceAuthToken = device_auth_token;
+  const dispatch = useDispatch();
+  const handlePageError = useErrorHandler();
+
+  const [showInfoModal, setShowInfoModal] = useState<boolean>(false);
+
+  const [refetchStartTime, setRefetchStartTime] = useState<number | null>(null);
+  const [showRefetchSpinner, setShowRefetchSpinner] = useState<boolean>(false);
+  const [hostSoftware, setHostSoftware] = useState<ISoftware[]>([]);
+  const [host, setHost] = useState<IHost | null>();
+  const [orgLogoURL, setOrgLogoURL] = useState<string>("");
+
+  const { data: deviceMapping, refetch: refetchDeviceMapping } = useQuery(
+    ["deviceMapping", deviceAuthToken],
+    () =>
+      deviceUserAPI.loadHostDetailsExtension(deviceAuthToken, "device_mapping"),
+    {
+      enabled: !!deviceAuthToken,
+      refetchOnMount: false,
+      refetchOnReconnect: false,
+      refetchOnWindowFocus: false,
+      retry: false,
+      select: (data: IDeviceMappingResponse) =>
+        data.device_mapping &&
+        data.device_mapping.filter(
+          (deviceUser) => deviceUser.email && deviceUser.email.length
+        ),
+    }
+  );
+
+  const refetchExtensions = () => {
+    deviceMapping !== null && refetchDeviceMapping();
+  };
+
+  const {
+    isLoading: isLoadingHost,
+    error: loadingDeviceUserError,
+    refetch: refetchHostDetails,
+  } = useQuery<IHostResponse, Error, IHostResponse>(
+    ["host", deviceAuthToken],
+    () => deviceUserAPI.loadHostDetails(deviceAuthToken),
+    {
+      enabled: !!deviceAuthToken,
+      refetchOnMount: false,
+      refetchOnReconnect: false,
+      refetchOnWindowFocus: false,
+      retry: false,
+      select: (data: IHostResponse) => data,
+      onSuccess: (returnedHost) => {
+        setShowRefetchSpinner(returnedHost.host.refetch_requested);
+        if (returnedHost.host.refetch_requested) {
+          // If the API reports that a Fleet refetch request is pending, we want to check back for fresh
+          // host details. Here we set a one second timeout and poll the API again using
+          // fullyReloadHost. We will repeat this process with each onSuccess cycle for a total of
+          // 60 seconds or until the API reports that the Fleet refetch request has been resolved
+          // or that the host has gone offline.
+          if (!refetchStartTime) {
+            // If our 60 second timer wasn't already started (e.g., if a refetch was pending when
+            // the first page loads), we start it now if the host is online. If the host is offline,
+            // we skip the refetch on page load.
```

Comment:
I know these comments were copied over, but we should consider removing them. Since the code is readable, it doesn't seem necessary to expound on what the code is doing here. My primary concern is that the code will change and the comments won't get updated, which will result in confusion down the road.